common/src/main/java/org/openmrs/analytics/FhirStoreUtil.java

```diff
@@ -124,12 +124,13 @@ protected IGenericClient createGenericClient(String sinkUrl, Collection<IClientI
 		
 		for (Bundle.BundleEntryComponent entry : bundle.getEntry()) {
 			Resource resource = entry.getResource();
-			
+			String[] sepUrl = resource.getId().split("/");
```

Comment:
it looked Obvious to me that `resource.getId()` would return the resource id here but it was returning the Full URL , probably its a bug in HAPI. 
so i jsut had to split the full url string and extract the resource id (uuid)

Reply:
@bashir2 , `getIdPart()`  is undefined for a resource object.

so you want me to do some like 

```
IdType type= new IdType(resource.getId());

component.getRequest().setUrl(resource.fhirType() + "/" + type.getIdPart()).setMethod(Bundle.HTTPVerb.PUT);
```

Reply:
Hmm, not quite. You should be able to get `IdType` from `resource` directly, e.g., something like `resource.getIdElement().getIdPart()`. Does that work?

Reply:
sure , that works , let me test it out and commit it

Reply:
sure Updated to getting idTYpe direvtly from resource. Sure it works pretty well